tests/main/mounts-persist-refresh-content-snap/task.yaml

```diff
@@ -35,26 +36,30 @@ prepare: |
   tests.cleanup defer tests.session -u test restore
 
 execute: |
+  # Construct the mount namespace once so that we are not racing construction
+  # from background job of snap run --shell with update with snap install of
+  # the test-snapd-content-slot below.
+  test-snapd-desktop-layout-with-content.sh -c 'true'
   touch /run/keep-running
-  # read a file continuously in the background until it fails - note that the
+  # Read a file continuously in the background until it fails - note that the
   # while loop here has to be inside the snap run shell, since the process must
   # persist during the refresh, if it is on the outside the mount namespace will
-  # be rebuilt and the crash will not be reproduced
-  tests.session -u test exec snap run --shell test-snapd-desktop-layout-with-content.cmd -c 'while test -f /run/keep-running && test -d /usr/share/fonts/foo-font; do true; done' &
+  # be rebuilt and the crash will not be reproduced.
+  tests.session -u test exec test-snapd-desktop-layout-with-content.sh -c \
+    'while test -f /run/keep-running && test -d /usr/share/fonts/foo-font; do true; done; ls -ld /usr/share/fonts/foo-font' &
```

Comment:
hm could we make this bit more robust with something like:
```
set -e
while test -f /run/keep-running; do
    if ! [ -d /usr/share/fonts/foo-font ] ; then 
        echo "foo-font missing" >> $SNAP_USER_COMMON/status
        exit 1
    fi
done
echo "exited" >> $SNAP_USER_COMMON/status
```
and then drop the ps -p check, proceed to remove the keep running flag, wait for exit and inspect contents of /home/test/.snap/test-snapd-desktop-layout/common/status

Reply:
Sure, let's do that! I wanted to limit the bug fixing but I agree that any embedded shell is tricky enough.